### PR TITLE
Switch-like if statement with misplaced curly brackets converted to switch statement

### PR DESCRIPTION
**Pull request by hasys**

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/main/java/org/kie/workbench/common/stunner/shapes/client/view/BasicShapeView.java [hunk 1]
```diff
@@ -322,24 +341,31 @@ public void destroy() {
...
-            } else {
-                eventHandlerManager.addHandler( type, eventHandler );
-            }
+        if (!supports(type)) {
+            return (T) this;
         }
-        return ( T ) this;
+
+        HandlerRegistration[] registrations;
+        switch (type) {
```
Comment (ederign) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/main/java/org/kie/workbench/common/stunner/shapes/client/view/BasicShapeView.java]: Why we don't place that logic in the enum?

Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/main/java/org/kie/workbench/common/stunner/shapes/client/view/BasicShapeView.java]: ViewEventType is enum or what do you mean?

**Comment by manstis** [comment]
@hasys @ederign Is this good to go? 

It looks like @ederign question about the enum change/use is outstanding.

**Comment by hasys** [comment]
Hi @manstis, this PR partially covered by https://github.com/droolsjbpm/kie-wb-common/pull/605 so I guess we can close it for now.

**Comment by manstis** [comment]
@hasys OK. Thanks for the clarification. Closing.

**Pull request closed without merging**